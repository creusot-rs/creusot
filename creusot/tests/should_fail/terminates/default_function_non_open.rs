extern crate creusot_contracts;
use creusot_contracts::*;

mod inner {
    use super::*;
    pub(super) trait Foo {
        #[open(self)]
        #[logic]
        fn f() {}
        #[logic]
        fn g();
    }
}

impl inner::Foo for i32 {
    #[open(self)]
    #[logic]
    fn g() {
        Self::f(); // this assumes f could call g
    }
}
