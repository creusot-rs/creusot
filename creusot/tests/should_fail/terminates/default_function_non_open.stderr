error: Mutually recursive functions: when calling `<impl inner::Foo for i32>::g`...
  --> default_function_non_open.rs:18:5
   |
18 |     fn g() {
   |     ^^^^^^
   |
note: then `<impl inner::Foo for i32>::g` might call `<impl inner::Foo for i32>::g` via the call to `inner::Foo::f`.
  --> default_function_non_open.rs:19:9
   |
19 |         Self::f(); // this assumes f could call g
   |         ^^^^^^^^^

error: aborting due to 1 previous error

